aws/sdk/integration-tests/s3/tests/identity-cache.rs

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+use std::sync::atomic::{AtomicI32, Ordering};
+use std::sync::Arc;
+
+use aws_config::{identity::IdentityCache, BehaviorVersion, Region};
+use aws_credential_types::{
+    provider::{future::ProvideCredentials as ProvideCredentialsFuture, ProvideCredentials},
+    Credentials,
+};
+use aws_sdk_s3::Client;
+use aws_smithy_runtime::client::http::test_util::infallible_client_fn;
+
+#[tokio::test]
+async fn test_identity_cache_reused_by_default() {
```

Comment:
If an integration test is service agnostic, it's a good practice we write it in Kotlin. Since the test uses `credentials_provider`, you can use `awsSdkIntegrationTest` ([example](https://github.com/smithy-lang/smithy-rs/blob/8bb11b8cc41d4c107d532c165ccc062c59780719/aws/sdk-codegen/src/test/kotlin/software/amazon/smithy/rustsdk/EndpointsCredentialsTest.kt#L81)).

Reply:
Does this end up on every generated client? If so is that what we want here? 

It does use `credentials_provider` but with fake/test credentials. It really is testing the cache behavior and interaction between `SdkConfig` and generated client config.

Reply:
I don't think it does. `awsSdkIntegrationTest` is a test helper that tests codegen itself but does not run against generated AWS SDKs.

> It really is just testing the cache behavior and interaction between SdkConfig and generated client config. It doesn't make a real request.

That does sound like service-agnostic, and that's what `awsSdkIntegrationTest` is for

Reply:
Just throwing one last thing on the table before leaving tests in `integration-tests/s3`, can we directly construct `SdkConfig` in tests with necessary fields without going through `aws_config::defaults`? I'm assuming the tests won't need default configs in order to verify what they are supposed to test? If we don't use `aws-config`, then we may be able to avoid the problem we discussed offline.

Reply:
We _can_ which would allow us to test some of this but not how different behavior versions behave in loading (which still require a subsequent PR to fully implement) if I understand correctly how it's all wired.